src/Microsoft.AspNet.Server.Kestrel/Http/SocketOutput.cs

```diff
@@ -388,6 +387,22 @@ void ISocketOutput.Write(ArraySegment<byte> buffer, bool immediate, bool chunk)
 
         Task ISocketOutput.WriteAsync(ArraySegment<byte> buffer, bool immediate, bool chunk, CancellationToken cancellationToken)
         {
+            if (cancellationToken.IsCancellationRequested)
```

Comment:
Done; not sure perf effects; but on the plus side I've removed some boxing and unboxing :stuck_out_tongue_winking_eye: